tls/s2n_connection.h

```diff
@@ -304,6 +304,7 @@ struct s2n_connection {
     s2n_session_ticket_status session_ticket_status;
     struct s2n_blob client_ticket;
     uint32_t ticket_lifetime_hint;
+    struct s2n_ticket_fields tls13_ticket_fields;
```

Comment:
Why are we moving this to the connection struct? I like that it makes calling functions more clean, but it does add like 48 bytes right?

Reply:
We're moving this to the connection structure so that the information is accessible to `s2n_connection_get_session.`

Doing this makes the connection object slightly larger, but the secret is heap memory and only allocates as much as it needs. The secret seems like the most reasonable thing to store: the full serialized state would take more memory, and the nonce potentially would too since it theoretically has a max size of UINT8_MAX.